base/util.go

```diff
@@ -1056,3 +1057,70 @@ func Sha1HashString(str string, salt string) string {
 	hashedKey := h.Sum(nil)
 	return fmt.Sprintf("%x", hashedKey)
 }
+
+// InjectJSONProperty takes the given JSON byte slice, marshals val, and inserts under the given key, and returns the result.
```

Comment:
Maybe include some comments/documentation here about implications of duplicate keys.